File: readability/readability.py
# Implements a program that computes the approximate grade level of text using the Coleman-Liau index
from cs50 import get_string


def main():

    # gets text from user
    text = get_string("Text: ")

    # counts letters, words, and sentences in text
    letters, words, sentences = count(text)

    # calculates grade level using Coleman-Liau formula
    grade = round(0.0588 * letters / words * 100.00 - 0.296 * sentences / words * 100.00 - 15.8)

    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Grade " + str(grade))


# counts number of letters, words, and spaces
def count(text):

    letters, words, sentences = 0, 1, 0

    for char in text:

        if char.isalpha() == True:
            letters += 1

        if char.isspace() == True:
            words += 1

        if "." in char or "!" in char or "?" in char:
            sentences += 1

    return letters, words, sentences


main()

File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    int length = 0, letters = 0, words = 1, sentences = 0, level;

    //prompts user for text
    string text = get_string("Text: ");
    length = strlen(text);

    for (int i = 0, n = length; i < n; i++)
    {

        // counts number of letters
        if (isalpha(text[i]) != 0)
        {
            letters++;
        }

        //counts number of words
        if (isspace(text[i]) != 0)
        {
            words++;
        }

        //counts number of sentences
        if (text[i] == 46 || text[i] == 33 || text[i] == 63)
        {
            sentences++;
        }
    }

    //determines grade level using Coleman-Liau index
    level = round(0.0588 * (float) letters / words * 100 - 0.296 * (float) sentences / words * 100 - 15.8);

    //prints grade levels below 1
    if (level < 1)
    {
        printf("Before Grade 1\n");
    }

    //prints grade levels above 16
    else if (level >= 16)
    {
        printf("Grade 16+\n");
    }

    //prints grade level
    else
    {
        printf("Grade %i\n", level);
    }
}
